runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java

```diff
@@ -151,6 +151,9 @@ private FileDescriptor getDescriptor(String name) throws DescriptorValidationExc
             return fileDescriptorMap.get(name);
         }
 
+        if (!fdProtoMap.containsKey(name)) {
```

Comment:
Is it possible to collapse this with fileDescriptorMap?

It's also seems a little bit strange to call getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, REGISTRY_TABLE_NAME) inside the constructor I feel like its cleaner to open that map and extract the serializers from a higher layer.

Reply:
I have been exploring this, looks like right there isn't an obvious way to directly convert from a type descriptor to a type. That is why we are having this extra layer of indirection to capture the File itself. It definitely needs more digging that a debugging patch.